go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -58,14 +78,17 @@ func NewMaster(kube kube.Interface) (*MasterController, error) {
 	return m, nil
 }
 
-// StartMaster creates and starts the hybrid overlay master controller
-func StartMaster(kube kube.Interface, wf *factory.WatchFactory) error {
-	klog.Infof("Starting hybrid overlay master...")
-	master, err := NewMaster(kube)
-	if err != nil {
-		return err
-	}
-	return houtil.StartNodeWatch(master, wf)
+// Run starts the controller
+func (m *MasterController) Run(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	klog.Info("Starting Hybrid Overlay Master Controller")
+
+	klog.Info("Starting workers")
+	_ = m.nodeInformer.Run(informer.DefaultNodeInformerThreadiness, stopCh)
```

Comment:
Do you need to do this? I thought that starting the SharedInformerFactory started all extant informers.

Reply:
The informers get started via the `factory.Start()` method, which is called by the WatchFactory or in this case, the code that creates the informers in `hybrid-overlay/cmd/hybrid-overlay-node/hybrid-overlay-node.go`. This code is calling [the informer wrapper's `Run` method](https://github.com/ovn-org/ovn-kubernetes/pull/1314/files/a3b0112765e2acc50f551d8bc1c0d8e8b93fda49#diff-b7d4a9b4c13348779a6d3e9b1092a268R77) from `pkg/informer` which starts the workers that process the workqueue.

So yes, it's needed.

Reply:
Aha, gotcha. Naming tripped me up a bit there. Perhaps `nodeInformer` should be called `nodeEventHandler`?